src/domain/fiat/display-currency.ts

```diff
@@ -25,5 +26,34 @@ export const majorToMinorUnit = ({
   displayMajorExponent: CurrencyMajorExponent
 }) => Number(Number(amount) * 10 ** displayMajorExponent)
 
-export const usdMajorToMinorUnit = (amount: number | bigint) =>
-  majorToMinorUnit({ amount, displayMajorExponent: MajorExponent.STANDARD })
+export const currencyMajorToMinorUnit = ({
+  amount,
+  displayCurrency,
+}: {
+  amount: number | bigint
+  displayCurrency: DisplayCurrency
+}) => {
+  const displayMajorExponent = getCurrencyMajorExponent(displayCurrency)
+  return majorToMinorUnit({ amount, displayMajorExponent })
+}
+
+export const getCurrencyMajorExponent = (
+  currency: DisplayCurrency,
+): CurrencyMajorExponent => {
+  try {
+    const formatter = new Intl.NumberFormat("en-US", { style: "currency", currency })
+    const { minimumFractionDigits } = formatter.resolvedOptions()
+    switch (minimumFractionDigits) {
+      case 0:
+        return MajorExponent.ZERO
+      case 1:
+        return MajorExponent.ONE
+      case 3:
+        return MajorExponent.THREE
+      default:
+        return MajorExponent.STANDARD
+    }
+  } catch {
+    return MajorExponent.STANDARD
```

Comment:
Silent catching is not good! Either this is a -we want it to crash- situation (ie remove try/catch) or we should be explicit about what we are catching and at least add it to the trace.

Reply:
so tracing service is an exception? I can add a comment but this is necessary for alias currencies (we dont want to crash to get the major exponent)